api/src/pcapi/core/educational/models.py

```diff
@@ -909,13 +909,19 @@ def cancel_booking(
         self.cancellationReason = reason
         self.dateUsed = None
 
-    def uncancel_booking_set_used(self) -> None:
+    def uncancel_booking(self) -> None:
         if not (self.status is CollectiveBookingStatus.CANCELLED):
             raise booking_exceptions.BookingIsNotCancelledCannotBeUncancelled()
         self.cancellationDate = None
         self.cancellationReason = None
         self.status = CollectiveBookingStatus.USED
-        self.dateUsed = datetime.utcnow()
+        if self.confirmationDate:
+            self.status = CollectiveBookingStatus.CONFIRMED
+        else:
+            self.status = CollectiveBookingStatus.PENDING
+        if self.collectiveStock.beginningDatetime < datetime.utcnow():
+            self.status = CollectiveBookingStatus.USED
+            self.dateUsed = datetime.utcnow()
```

Comment:
```suggestion
        if self.confirmationDate:
            self.status = CollectiveBookingStatus.CONFIRMED
            if self.collectiveStock.beginningDatetime < datetime.utcnow():
                 self.status = CollectiveBookingStatus.USED
                 self.dateUsed = datetime.utcnow()
             else:
                 self.status = CollectiveBookingStatus.CONFIRMED
        else:
            self.status = CollectiveBookingStatus.PENDING

```pour êter certain de toujours mettre un status